### PR TITLE
docs: add GOVERNANCE doc

### PR DESCRIPTION
**Pull request by feng-tao**
### Summary of Changes
Add GOVERNANCE model based on envoy governance doc (https://github.com/envoyproxy/envoy/blob/master/GOVERNANCE.md)

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ ] PR title addresses the issue accurately and concisely, including a title prefix.
- [ ] PR includes a summary of changes.
- [ ] My commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"

### CODE REVIEW

**Comment by feng-tao** [comment]
cc @lyft/amundsen-dev

File: GOVERNANCE.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Process for becoming a maintainer
+
+## Your organization is not yet a maintainer
```
Comment (Golodhros) [inline GOVERNANCE.md:3]:
```suggestion
## If your organization is not yet a maintainer
```

File: GOVERNANCE.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+* Express interest to the senior maintainers that your organization is interested in becoming a maintainer. Becoming a maintainer generally means that you are going to be spending substantial time (>25%) on Amundsen for the foreseeable future. Ultimately your goal is to become a senior maintainer that will represent your organization.
+* We will expect you to start contributing increasingly complicated PRs, under the guidance of the existing senior maintainers.
+* We may ask you to do some PRs from our backlog.
+* As you gain experience with the code base and our standards, we will ask you to do code reviews for incoming PRs (i.e., all maintainers are expected to shoulder a proportional share of community reviews).
+* After a period of approximately 2-3 months of working together and making sure we see eye to eye, the existing senior maintainers will confer and decide whether to grant maintainer status or not. We make no guarantees on the length of time this will take, but 2-3 months is the approximate goal.
+
+## Your organization is currently a maintainer
```
Comment (Golodhros) [inline GOVERNANCE.md:11]:
```suggestion
## If your organization is currently a maintainer
```

File: GOVERNANCE.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+* First decide whether your organization really needs more people with maintainer access. Valid reasons are "blast radius", a large organization that is working on multiple unrelated projects, etc.
+* Contact a senior maintainer for your organization and express interest.
+* Start doing PRs and code reviews under the guidance of your senior maintainer.
+* After a period of 1-2 months the existing senior maintainers will discuss granting "standard" maintainer access. This will require 3 (senior) maintainers +1 vote and no -1 vote.
+* "Standard" maintainer access can be upgraded to "senior" maintainer access after another 1-2 months of work and another conference of the existing senior committers.
+
+## Maintainer responsibilities
```
Comment (Golodhros) [inline GOVERNANCE.md:19]:
```suggestion
# Maintainer responsibilities
```

File: GOVERNANCE.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+* We will expect you to start contributing increasingly complicated PRs, under the guidance of the existing senior maintainers.
+* We may ask you to do some PRs from our backlog.
+* As you gain experience with the code base and our standards, we will ask you to do code reviews for incoming PRs (i.e., all maintainers are expected to shoulder a proportional share of community reviews).
+* After a period of approximately 2-3 months of working together and making sure we see eye to eye, the existing senior maintainers will confer and decide whether to grant maintainer status or not. We make no guarantees on the length of time this will take, but 2-3 months is the approximate goal.
+
+## Your organization is currently a maintainer
+
+* First decide whether your organization really needs more people with maintainer access. Valid reasons are "blast radius", a large organization that is working on multiple unrelated projects, etc.
```
Comment (Golodhros) [inline GOVERNANCE.md:13]:
```suggestion
* First, decide whether your organization needs more people with maintainer access. Valid reasons are "blast radius", a large organization working on multiple unrelated projects, etc.
```

File: GOVERNANCE.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
+# Process for becoming a maintainer
+
+## Your organization is not yet a maintainer
+
+* Express interest to the senior maintainers that your organization is interested in becoming a maintainer. Becoming a maintainer generally means that you are going to be spending substantial time (>25%) on Amundsen for the foreseeable future. Ultimately your goal is to become a senior maintainer that will represent your organization.
+* We will expect you to start contributing increasingly complicated PRs, under the guidance of the existing senior maintainers.
+* We may ask you to do some PRs from our backlog.
+* As you gain experience with the code base and our standards, we will ask you to do code reviews for incoming PRs (i.e., all maintainers are expected to shoulder a proportional share of community reviews).
+* After a period of approximately 2-3 months of working together and making sure we see eye to eye, the existing senior maintainers will confer and decide whether to grant maintainer status or not. We make no guarantees on the length of time this will take, but 2-3 months is the approximate goal.
```
Comment (Golodhros) [inline GOVERNANCE.md:9]:
```suggestion
* After approximately 2-3 months of working together and making sure we see eye to eye, the existing senior maintainers will confer and decide whether to grant maintainer status or not. We make no guarantees on the length of time this will take, but 2-3 months is the approximate goal.
```

File: GOVERNANCE.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
+# Process for becoming a maintainer
+
+## Your organization is not yet a maintainer
+
+* Express interest to the senior maintainers that your organization is interested in becoming a maintainer. Becoming a maintainer generally means that you are going to be spending substantial time (>25%) on Amundsen for the foreseeable future. Ultimately your goal is to become a senior maintainer that will represent your organization.
```
Comment (Golodhros) [inline GOVERNANCE.md:5]:
```suggestion
* Start by expressing interest to the senior maintainers that your organization is interested in becoming a maintainer. Becoming a maintainer generally means that you are going to be spending substantial time (>25%) on Amundsen for the foreseeable future. Ultimately your goal is to become a senior maintainer that will represent your organization.
```

File: GOVERNANCE.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+* We may ask you to do some PRs from our backlog.
+* As you gain experience with the code base and our standards, we will ask you to do code reviews for incoming PRs (i.e., all maintainers are expected to shoulder a proportional share of community reviews).
+* After a period of approximately 2-3 months of working together and making sure we see eye to eye, the existing senior maintainers will confer and decide whether to grant maintainer status or not. We make no guarantees on the length of time this will take, but 2-3 months is the approximate goal.
+
+## Your organization is currently a maintainer
+
+* First decide whether your organization really needs more people with maintainer access. Valid reasons are "blast radius", a large organization that is working on multiple unrelated projects, etc.
+* Contact a senior maintainer for your organization and express interest.
+* Start doing PRs and code reviews under the guidance of your senior maintainer.
+* After a period of 1-2 months the existing senior maintainers will discuss granting "standard" maintainer access. This will require 3 (senior) maintainers +1 vote and no -1 vote.
```
Comment (Golodhros) [inline GOVERNANCE.md:16]:
```suggestion
* After 1-2 months, the existing senior maintainers will discuss granting "standard" maintainer access. The approval requires 3 (senior) maintainers +1 vote and no -1 vote.
```

File: GOVERNANCE.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+* "Standard" maintainer access can be upgraded to "senior" maintainer access after another 1-2 months of work and another conference of the existing senior committers.
+
+## Maintainer responsibilities
+
+* Monitor email aliases.
+* Monitor Slack (delayed response is perfectly acceptable).
+* Triage GitHub issues and perform pull request reviews for other maintainers and the community.
+* Triage build issues - file issues for known flaky builds or bugs, and either fix or find someone to fix any master build breakages.
```
Comment (Golodhros) [inline GOVERNANCE.md:24]:
```suggestion
* Triage build issues - file issues for known flaky builds or bugs, fix or find someone to fix any master build breakages.
```

File: GOVERNANCE.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+* During GitHub issue triage, apply all applicable [labels](https://github.com/lyft/amundsen/labels) to each new issue. Labels are extremely useful for future issue follow up. Which labels to apply is somewhat subjective so just use your best judgment.
```
Comment (Golodhros) [inline GOVERNANCE.md:25]:
```suggestion
* During GitHub issue triage, apply all applicable [labels](https://github.com/lyft/amundsen/labels) to each new issue. Labels are extremely useful for future issue follow-ups. Which labels to apply is somewhat subjective, so use your best judgment.
```

File: GOVERNANCE.md [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+* Make sure that ongoing PRs are moving forward at the right pace or closing them.
+* In general continue to be willing to spend at least 25% of ones time working on Amundsen (~1.25) business days per week).
```
Comment (Golodhros) [inline GOVERNANCE.md:27]:
```suggestion
* In general, continue to be willing to spend at least 25% of one's time working on Amundsen (~1.25) business days per week).
```

File: GOVERNANCE.md [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+
+
+## When does a maintainer lose maintainer status
+
+If a maintainer is no longer interested or cannot perform the maintainer duties listed above, they should volunteer to be moved to emeritus status. In extreme cases this can also occur by a vote of the maintainers per the voting process below.
```
Comment (Golodhros) [inline GOVERNANCE.md:32]:
```suggestion
If a maintainer is no longer interested or cannot perform the maintainer duties listed above, they should volunteer to be moved to emeritus status. In extreme cases, this can also occur by a vote of the maintainers per the voting process below.
```

File: GOVERNANCE.md [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+
+
+## Conflict resolution and voting
+
+In general, we prefer that technical issues and maintainer membership are amicably worked out between the persons involved. If a dispute cannot be decided independently, the maintainers can be called in to decide an issue. If the maintainers themselves cannot decide an issue, the issue will be resolved by voting. The voting process is a simple majority in which each senior maintainer receives two votes and each normal maintainer receives one vote (we could introduce senior maintainer like PMC later).
```
Comment (Golodhros) [inline GOVERNANCE.md:37]:
```suggestion
In general, we prefer that technical issues and maintainer membership are amicably worked out between the involved persons. If a dispute cannot be decided independently, the maintainers can be called in to decide an issue. If the maintainers themselves cannot decide on a problem, the issue will be resolved by voting. The voting process is a simple majority in which each senior maintainer receives two votes, and each regular maintainer receives one vote (we could introduce senior maintainer like PMC later).
```

File: GOVERNANCE.md [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+
+## Adding new projects to the amundsen-io GitHub organization
```
Comment (Golodhros) [inline GOVERNANCE.md:39]:
```suggestion
# Adding new projects to the Amundsen-io GitHub organization
```

File: GOVERNANCE.md [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+
+New projects will be added to the amundsen-io organization via GitHub issue discussion in one of the existing projects in the organization. Once sufficient discussion has taken place (~3-5 business days but depending on the volume of conversation), the maintainers of *the project where the issue was opened* (since different projects in the organization may have different maintainers) will decide whether the new project should be added. See the section above on voting if the maintainers
```
Comment (Golodhros) [inline GOVERNANCE.md:41]:
```suggestion
We will add new projects to the Amundsen organization via GitHub issue discussion in one of the organization's existing projects. Once sufficient discussion has taken place (~3-5 business days but depending on the volume of conversation), the maintainers of *the project where the issue was opened* (since different projects in the organization may have different maintainers) will decide whether the new project should be added. See the section above on voting if the maintainers
```

File: GOVERNANCE.md [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+cannot easily decide.
+
+## References
```
Comment (Golodhros) [inline GOVERNANCE.md:44]: Is this a Glossary instead?

File: GOVERNANCE.md [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+- Senior maintainer == PMC in ASF
```
Comment (Golodhros) [inline GOVERNANCE.md:45]: What is ASF?

File: GOVERNANCE.md [hunk 17]
```diff
@@ -0,0 +1,46 @@
+# Process for becoming a maintainer
+
```
Comment (Golodhros) [inline GOVERNANCE.md:2]:
```suggestion
We have two different processes to welcome new maintainers to the project, depending if their organization is already a maintainer of Amundsen or not.
```

File: GOVERNANCE.md [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+* Make sure that ongoing PRs are moving forward at the right pace or closing them.
+* In general continue to be willing to spend at least 25% of ones time working on Amundsen (~1.25) business days per week).
+
+
+## When does a maintainer lose maintainer status
+
+If a maintainer is no longer interested or cannot perform the maintainer duties listed above, they should volunteer to be moved to emeritus status. In extreme cases this can also occur by a vote of the maintainers per the voting process below.
+
+
+## Conflict resolution and voting
```
Comment (Golodhros) [inline GOVERNANCE.md:35]: I would shuffle a bit the structure of this doc, with an outline like this:

- Conflict resolution
- Maintainer Responsibilities
- How to become a maintainer
- When a maintainer loses status
- Adding projects

**Comment by Golodhros** [comment]
This also relates to the draft we have for maintaining: https://docs.google.com/document/d/1ot41rVhFMVAFSCWVKfO6yNVwnnoXoJxeaXMoPEq28NM/edit#heading=h.dp6gygxquc8s right?

File: GOVERNANCE.md (hunk 4 above)
Comment (Alagappan) [reply GOVERNANCE.md:13]: Do you have any guidelines on how many we would like per organization on average? How many is too many? Are there other valid reasons than "blast radius" ?

File: GOVERNANCE.md [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+* We will expect you to start contributing increasingly complicated PRs, under the guidance of the existing senior maintainers.
+* We may ask you to do some PRs from our backlog.
+* As you gain experience with the code base and our standards, we will ask you to do code reviews for incoming PRs (i.e., all maintainers are expected to shoulder a proportional share of community reviews).
+* After a period of approximately 2-3 months of working together and making sure we see eye to eye, the existing senior maintainers will confer and decide whether to grant maintainer status or not. We make no guarantees on the length of time this will take, but 2-3 months is the approximate goal.
+
+## Your organization is currently a maintainer
+
+* First decide whether your organization really needs more people with maintainer access. Valid reasons are "blast radius", a large organization that is working on multiple unrelated projects, etc.
+* Contact a senior maintainer for your organization and express interest.
```
Comment (Alagappan) [inline GOVERNANCE.md:14]: What if your organization doesn't yet have a senior maintainer? Also, How would one know who is a senior maintainer and who is not? And how would they contact them? Is there a formal process to contact or can someone just DM directly on slack?

File: GOVERNANCE.md (hunk 7 above)
Comment (Alagappan) [reply GOVERNANCE.md:16]: Should be `atleast` 3 senior maintainers approval right? Also, could you elaborate on how this would be carried out? Should the member apply to be a maintainer or is this more a decision by senior maintainer to initiate the process?

Comment (Alagappan) [reply GOVERNANCE.md:16]: Would the approval conversation happen over a email thread or would it be a poll? Do you think we should add some guidelines on how long should the poll be open (like min of 3 business or 1 week) as the poll cannot be closed as soon as we get 3 approvals right? We would have to provide sufficient time for all senior members to review ( and potentially do a -1 if required).

File: GOVERNANCE.md (hunk 9 above)
Comment (Alagappan) [inline GOVERNANCE.md:25]:
```suggestion
* During GitHub issue triage, apply all applicable [labels](https://github.com/lyft/amundsen/labels) to each new issue. Labels are extremely useful for future issue follow up. Adding labels is somewhat subjective so please use your best judgement.
```

File: GOVERNANCE.md (hunk 14 above)
Comment (Alagappan) [reply GOVERNANCE.md:41]: Not clear what `new projects` here refers to. Do you mean adding `new features to Amundsen` and creating an issue in one of the sub-components?

File: GOVERNANCE.md (hunk 7 above)
Comment (Alagappan) [reply GOVERNANCE.md:16]: could you elaborate on what is required for someone to vote -1? Also, could you elaborate on what is the process when someone is denied maintained status? Should they expect some communication around what they should next?

File: GOVERNANCE.md (hunk 5 above)
Comment (Alagappan) [reply GOVERNANCE.md:9]: Do we need to say `making sure we eye to eye`? I would just rephrase this to read `After about 2-3 months of working on the project, an existing senior maintainer will initiate a voting process to decide whether to award maintainer status. Please note 2-3 months is just a suggested timeline and should by no means considered a guarantee to get maintainer status. Each contributor's timeline may vary widely based on the number and complexity of contributions` -  Please let me know what you think?

**Comment by feng-tao** [comment]
this pr is more for a reference and not the actual one.

File: GOVERNANCE.md [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+* Start doing PRs and code reviews under the guidance of your senior maintainer.
```
Comment (Alagappan) [inline GOVERNANCE.md:15]:
```suggestion
* Start by resolving [open issues](https://github.com/lyft/amundsen/issues), contributing code changes and/or reviewing open pull requests.
```

File: GOVERNANCE.md (hunk 14 above)
Comment (Golodhros) [reply GOVERNANCE.md:41]: I think this means new repositories on the organization.

File: GOVERNANCE.md (hunk 19 above)
Comment (Golodhros) [reply GOVERNANCE.md:14]: I think those are legit questions.
